src/core/slice.c

```diff
@@ -84,6 +84,7 @@ static int slice_add_default_dependencies(Slice *s) {
 
 static int slice_verify(Slice *s) {
         _cleanup_free_ char *parent = NULL;
+        bool check_recursive_units = UNIT(s)->manager && FLAGS_SET(UNIT(s)->manager->test_run_flags, MANAGER_TEST_RUN_IGNORE_DEPENDENCIES);
```

Comment:
the naming is reversed, and also it's dereferencing before the asserts. Just do the check inline down below, without an additional bool variable